src/components/annotations/draw_arrow_head.js

```diff
@@ -23,110 +22,116 @@ var ARROWPATHS = require('./arrow_paths');
-            if(backOff) dashArray = '0px,' + backOff + 'px,';
-        }
-
-        if(doEnd) {
-            var end0 = el.getPointAtLength(pathlen),
-                dend = el.getPointAtLength(pathlen - 0.1);
-            endRot = Math.atan2(end0.y - dend.y, end0.x - dend.x);
-            end = el.getPointAtLength(Math.max(0, pathlen - backOff));
-
-            if(backOff) {
-                var shortening = dashArray ? 2 * backOff : backOff;
-                dashArray += (pathlen - shortening) + 'px,' + pathlen + 'px';
-            }
-        }
-        else if(dashArray) dashArray += pathlen + 'px';
-
-        if(dashArray) el3.style('stroke-dasharray', dashArray);
+  } else if (el.nodeName === 'path') {
+    var pathlen = el.getTotalLength(),
+      // using dash to hide the backOff region of the path.
+      // if we ever allow dash for the arrow we'll have to
+      // do better than this hack... maybe just manually
+      // combine the two
+      dashArray = '';
+
+    if (pathlen < backOff) {
+      hideLine();
+      return;
     }
 
-    function hideLine() { el3.style('stroke-dasharray', '0px,100px'); }
-
-    function drawhead(p, rot) {
-        if(!headStyle.path) return;
-        if(style > 5) rot = 0; // don't rotate square or circle
-        d3.select(el.parentElement).append('path')
-            .attr({
-                'class': el3.attr('class'),
-                d: headStyle.path,
-                transform:
-                    'translate(' + p.x + ',' + p.y + ')' +
-                    'rotate(' + (rot * 180 / Math.PI) + ')' +
-                    'scale(' + scale + ')'
-            })
-            .style({
-                fill: stroke,
-                opacity: opacity,
-                'stroke-width': 0
-            });
+    if (doStart) {
+      var start0 = el.getPointAtLength(0), dstart = el.getPointAtLength(0.1);
+      startRot = Math.atan2(start0.y - dstart.y, start0.x - dstart.x);
+      start = el.getPointAtLength(Math.min(backOff, pathlen));
+      if (backOff) dashArray = '0px,' + backOff + 'px,';
     }
 
-    if(doStart) drawhead(start, startRot);
-    if(doEnd) drawhead(end, endRot);
+    if (doEnd) {
+      var end0 = el.getPointAtLength(pathlen),
+        dend = el.getPointAtLength(pathlen - 0.1);
+      endRot = Math.atan2(end0.y - dend.y, end0.x - dend.x);
+      end = el.getPointAtLength(Math.max(0, pathlen - backOff));
+
+      if (backOff) {
+        var shortening = dashArray ? 2 * backOff : backOff;
+        dashArray += pathlen - shortening + 'px,' + pathlen + 'px';
+      }
+    } else if (dashArray) dashArray += pathlen + 'px';
+
+    if (dashArray) el3.style('stroke-dasharray', dashArray);
+  }
+
+  function hideLine() {
+    el3.style('stroke-dasharray', '0px,100px');
+  }
+
+  function drawhead(p, rot) {
+    if (!headStyle.path) return;
+    if (style > 5) rot = 0; // don't rotate square or circle
+    d3
+      .select(el.parentElement)
+      .append('path')
```

Comment:
This one is going to be tricky.

I'd be surprised if `prettier` respects `d3` half-indenting conventions.

:+1: to get rid of half-indents.
:-1: for not respecting vertical formatting